devicetypes/inovelliusa/inovelli-bulb-multi-white.src/inovelli-bulb-multi-white.groovy

```diff
@@ -62,33 +68,70 @@ metadata {
 	details(["switch", "levelSliderControl", "colorTempSliderControl", "refresh"])
 }
 
+private getCOLOR_TEMP_MIN() { 2700 }
+private getCOLOR_TEMP_MAX() { 6500 }
 private getWARM_WHITE_CONFIG() { 0x51 }
 private getCOLD_WHITE_CONFIG() { 0x52 }
 private getWARM_WHITE() { "warmWhite" }
 private getCOLD_WHITE() { "coldWhite" }
 private getWHITE_NAMES() { [WARM_WHITE, COLD_WHITE] }
+private getZWAVE_COLOR_COMPONENT_ID() { [warmWhite: 0, coldWhite: 1, red: 2, green: 3, blue: 4] }
+private getCOLOR_TEMP_DIFF() { COLOR_TEMP_MAX - COLOR_TEMP_MIN }
+
 
 def updated() {
 	log.debug "updated().."
+    if (!state.powerStateMem) initializeVars()
+    if (!state.bulbMemory) initializeVars()
+	if (state.powerStateMem.toInteger() != bulbMemory.toInteger())
+	if (logEnable) runIn(1800,logsOff)
+    configure()
 	response(refresh())
 }
 
+def initializeVars() {
+	if (!state.colorReceived) state.colorReceived = [red: null, green: null, blue: null, warmWhite: null, coldWhite: null]
+	if (!state.powerStateMem) state.powerStateMem=0
+}
+
+def logsOff(){
+    log.warn "debug logging disabled..."
+    device.updateSetting("logEnable",[value:"false",type:"bool"])
+}
+
 def installed() {
 	log.debug "installed()..."
 	sendEvent(name: "checkInterval", value: 1860, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "0"])
 	sendEvent(name: "level", value: 100, unit: "%")
 	sendEvent(name: "colorTemperature", value: 2700)
+    initializeVars()
+}
+
+def configure() {
+	def cmds = []
+    if (!bulbMemory) bulbMemor=0
```

Comment:
typo in assignment (missing `y`)

I don't know groovy well at all, but I'm also wondering if these need to refer to `state.bulbMemory` instead of just `bulbMemory`? or is the object implied here for some reason?

Reply:
Thanks for the catch... and it relates to the device preferences .